robot/impl/local_robot.go

```diff
@@ -656,6 +656,12 @@ func (r *localRobot) newResource(
 	return res, nil
 }
 
+func (r *localRobot) updateWeakDependentsInLock(ctx context.Context) {
+	r.manager.resourceGraphLock.Lock()
+	defer r.manager.resourceGraphLock.Unlock()
+	r.updateWeakDependents(ctx)
+}
+
 func (r *localRobot) updateWeakDependents(ctx context.Context) {
```

Comment:
Why not lock and defer unlock of `resourceGraphLock` at the top of this method instead of introducing the `InLock` method? Are you trying _not_ to lock when calling from `completeConfig` in the resource manager (call `updateWeakDependents` without a lock? If so, why?

Reply:
completeConfig already holds a resourceGraphLock so didn't think we should grab it again

Reply:
Cool; that's sort of what I assumed. I don't have a super strong opinion, but I would've thought that the `completeConfig` call to weak dependents would be the one that's "abnormal" in terms of method name: something like `updateWeakDependentsWithoutLock`, while the other call sites are just `updateWeakDependents` and take the lock.

But also, I'm more broadly concerned with just adding this locking in the first place, so see my other comment.